### PR TITLE
Refactor OS plugin to incorporate OCI OS UploadManager for enhanced  functionality.

### PR DESCRIPTION
**Pull request by mrDzurb**
## Description
https://jira.oci.oraclecorp.com/browse/ODSC-43644
Refactors OS plugin to incorporate OCI OS [UploadManager](https://docs.oracle.com/en-us/iaas/tools/python/2.104.1/api/upload_manager.html) for uploading artifacts. The change enhancing functionality and improving performance.

## Tests
image
image

### CODE REVIEW

**Comment by mrDzurb** [comment]
image

**Comment by mrDzurb** [comment]
image

**Comment by mrDzurb** [comment]
image
image
image

**Review by darenr — APPROVED** [review]

File: oci_mlflow/oci_object_storage.py [hunk 1]
```diff
@@ -49,43 +110,77 @@ def _download_file(self, remote_file_path, local_path):
...
             for f in filenames:
-                self._upload_file(
-                    local_file=os.path.join(root, f),
-                    dest_path=os.path.join(upload_path, f),
+                artifact_uploader.upload(
+                    file_path=os.path.join(root, f),
+                    dst_path=os.path.join(upload_path, f),
                 )
 
     def get_fs(self):
```
Comment (mayoor) [inline oci_mlflow/oci_object_storage.py:159]: We can drop this method if it is not used anywhere

**Comment by mayoor** [comment]
> image image image

For completeness could you please test with a LLM model with >= 7b parameters?

File: oci_mlflow/oci_object_storage.py (hunk 1 above)
Comment (mrDzurb) [reply oci_mlflow/oci_object_storage.py:159]: It is used in `list_artifacts` and `delete_artyifacts` methods.

**Review by mayoor — APPROVED** [review]

**Pull request merged**